satsense/features/sift.py

```diff
@@ -5,33 +5,50 @@
 import numpy as np
 from sklearn.cluster import MiniBatchKMeans
 
+from ..generators import FullGenerator
 from ..image import Image
 from .feature import Feature
 
-SIFT = cv2.xfeatures2d.SIFT_create()
-
 
-def sift_cluster(images: Iterator[Image], n_clusters=32,
-                 sample_size=100000) -> MiniBatchKMeans:
+def sift_cluster(images: Iterator[Image],
+                 n_clusters=32,
+                 max_samples=100000,
+                 sample_window=(8192, 8192)) -> MiniBatchKMeans:
     """Create the clusters needed to compute the sift feature."""
-    descriptors = None
+    nfeatures = int(max_samples / len(images))
+    sift_object = cv2.xfeatures2d.SIFT_create(nfeatures)
```

Comment:
Use `sift_object = cv2.xfeatures2d.SIFT_create(max_features_per_window)` and remove the random choice of features below?